### DISCUSSION TITLE
RPi 3B - Targeting a specific port affects other ports as well

### DISCUSSION THREAD

**Discussion by Gil80** [mvp/uhubctl#421] [Q&A] [answered]
Hi,
I'm using this tool to toggle on/off my USB power LED strip on my Raspberry Pi 3B which is connected to my 3D printer.

The USB hub has a webcam connected to it. Each time I issue a command like  `uhubctl -l 1-1 -p 2 -a 0`
the webcam stream gets stuck and I have to reboot my RPi.

Is there a way to avoid this?

**Comment by mvp** [comment, marked as answer]
This is FAQ: https://github.com/mvp/uhubctl#raspberry-pi-turns-power-off-on-all-ports-not-just-the-one-i-specified

## Raspberry Pi turns power off on all ports, not just the one I specified

This is the limitation of Raspberry Pi hardware design. As a workaround, you can buy any external USB hub from supported list above, attach it to any USB port of Raspberry Pi, and control power on its ports independently. Also, there are supported hubs designed specifically for Raspberry Pi, e.g. UUGear MEGA4.
